test/interactions/clickInteractionTests.ts

```diff
@@ -1,160 +1,158 @@
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let c = new Plottable.Component();
-      c.renderTo(svg);
-
-      let clickInteraction = new Plottable.Interactions.Click();
-      clickInteraction.attachTo(c);
-
-      let callbackCalled = false;
-      let callback = () => callbackCalled = true;
-      clickInteraction.onClick(callback);
-
-      TestMethods.triggerFakeTouchEvent("touchstart", c.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-      TestMethods.triggerFakeTouchEvent("touchcancel", c.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-      TestMethods.triggerFakeTouchEvent("touchend", c.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-      assert.isFalse(callbackCalled, "callback not called since click was interrupted");
-
-      svg.remove();
+  describe("Click Interaction", () => {
+    describe("Basic Usage", () => {
+      let SVG_WIDTH = 400;
+      let SVG_HEIGHT = 400;
+
+      let svg: d3.Selection<void>;
+      let component: Plottable.Component;
+      let clickInteraction: Plottable.Interactions.Click;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        component = new Plottable.Component();
+        component.renderTo(svg);
+
+        clickInteraction = new Plottable.Interactions.Click();
+        clickInteraction.attachTo(component);
+      });
+
+      it("can register interaction using onClick()", () => {
+        let callbackCalled = false;
+        let lastPoint: Plottable.Point;
+        let callback = function(point: Plottable.Point) {
+          callbackCalled = true;
+          lastPoint = point;
+        };
+        clickInteraction.onClick(callback);
+
+        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+        assert.isTrue(callbackCalled, "callback called on clicking Component without moving mouse");
+        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2 }, "was passed correct point");
+
+        callbackCalled = false;
+        lastPoint = null;
+        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH / 4, SVG_HEIGHT / 4);
+        assert.isTrue(callbackCalled, "callback called on clicking and releasing inside the component");
+        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4 }, "was passed mouseup point");
+
+        callbackCalled = false;
+        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), SVG_WIDTH / 2, SVG_HEIGHT / 2);
+        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), SVG_WIDTH * 2, SVG_HEIGHT * 2);
+        assert.isFalse(callbackCalled, "callback not called if released outside component");
```

Comment:
"callback not called if mouse is released outside Component" makes the behavior being tested more clear.